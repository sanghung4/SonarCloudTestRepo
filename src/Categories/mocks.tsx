import { DocumentNode } from 'graphql';
import {
  ProductCategoriesDocument,
  ProductCategoriesQuery
} from 'generated/graphql';

export const success: MockCategory = {
  request: {
    query: ProductCategoriesDocument,
    variables: {
      engines: 'plumbing_hvac'
    }
  },
  result: {
    data: {
      productCategories: {
        categories: [
          {
            name: 'Faucets, Fixtures & Appliances',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Faucets, Showers & Bathroom Accessories',
                children: [
                  {
                    name: 'Showers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bathroom Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tub & Shower',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bathroom Faucets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Kitchen Faucets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tub Filler',
                    __typename: 'Category'
                  },
                  {
                    name: 'Spare Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Kitchen Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Appliances',
                children: [
                  {
                    name: 'Refrigerators And Freezers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hoods and Vents',
                    __typename: 'Category'
                  },
                  {
                    name: 'Outdoor Grills and BBQ',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ranges and Stoves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Microwaves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Garbage Disposals',
                    __typename: 'Category'
                  },
                  {
                    name: 'Appliance Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dishwashers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dryers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ovens',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Toilets, Urinals, Sinks & Flush Valves',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Toilet & Flush Valve Spare Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Manual Flush Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Spare Parts',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Fans & Ventilation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools',
                children: [],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Plumbing Installation, Tools, Hardware & Safety',
            children: [
              {
                name: 'Plumbing Installation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Clamps',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'Faucet and Toilet Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gaskets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Hangers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Riser',
                    __typename: 'Category'
                  },
                  {
                    name: 'Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'Anchors and Settings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Compression Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Power Tool',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brushes',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tool',
                    __typename: 'Category'
                  },
                  {
                    name: 'Shovels',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Hardware',
                children: [
                  {
                    name: 'Screws',
                    __typename: 'Category'
                  },
                  {
                    name: 'Washers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Nuts',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bolts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Hangers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Nails',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Safety',
                children: [
                  {
                    name: 'Gloves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Glasses',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hard Hat',
                    __typename: 'Category'
                  },
                  {
                    name: 'Vest',
                    __typename: 'Category'
                  },
                  {
                    name: 'Shoe Covers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Beanie',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ear Plugs',
                    __typename: 'Category'
                  },
                  {
                    name: 'Face Shield',
                    __typename: 'Category'
                  },
                  {
                    name: 'Knee Pads',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools, Hardware & Safety',
                children: [
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brushes',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Appliances',
                children: [
                  {
                    name: 'Garbage Disposals',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dishwashers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Washing Machines',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Pipe Accessories',
                children: [
                  {
                    name: 'Pipe Insulation',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'HVAC',
                children: [
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Discontinued',
                children: [],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Pipe & Fittings',
            children: [
              {
                name: 'Metal Pipe & Fittings',
                children: [
                  {
                    name: 'Copper Tube & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Steel Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Cast Iron Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brass Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ductile Iron Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plastic Pipe & Fittings - Supply',
                children: [
                  {
                    name: 'PEX Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'CPVC Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plastic Pipe & Fittings - Drainage',
                children: [
                  {
                    name: 'PVC Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'ABS Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plumbing Installation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Heating & Cooling',
            children: [
              {
                name: 'Air Distribution',
                children: [
                  {
                    name: 'Sheet Metal Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Grills, Registers & Diffusers',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Flex Duct',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sheet Metal Duct Pipe and Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Access & Inspection Panels',
                    __typename: 'Category'
                  },
                  {
                    name: 'CEILING RADIATION DAMPERS',
                    __typename: 'Category'
                  },
                  {
                    name: 'Grills, Registers, & Diffusers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Repair & Replacement Parts',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Capacitors',
                    __typename: 'Category'
                  },
                  {
                    name: 'OEM Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Contactors, Relays, Transformers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Motors',
                    __typename: 'Category'
                  },
                  {
                    name: 'Driers & Sight Glasses',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Residential Equipment',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'AC Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'HP Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Coils',
                    __typename: 'Category'
                  },
                  {
                    name: 'Furnaces',
                    __typename: 'Category'
                  },
                  {
                    name: 'Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'furnaces',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Commercial Equipment',
                children: [
                  {
                    name: 'Commercial Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Commercial Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Geothermal',
                    __typename: 'Category'
                  },
                  {
                    name: 'Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Installation Supplies',
                children: [
                  {
                    name: 'Tapes, Silicon, & Adhesives',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electrical',
                    __typename: 'Category'
                  },
                  {
                    name: 'Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensate Management',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Fans & Ventilation',
                children: [
                  {
                    name: 'Gas Vent Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ventilation Fans',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'IAQ',
                children: [
                  {
                    name: 'Air Filters',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Cleaners',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dehumidifiers',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Humidifiers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Controls, Sensors, & Zoning',
                children: [
                  {
                    name: 'Thermostats',
                    __typename: 'Category'
                  },
                  {
                    name: 'Zoning',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Ductless',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Hot Water, Valves, Irrigation & Pumps',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heater Parts & Accessories',
                children: [
                  {
                    name: 'Venting',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Heater Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Commercial Conversion Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Burners',
                    __typename: 'Category'
                  },
                  {
                    name: 'Heating Element',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gas Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Switches',
                    __typename: 'Category'
                  },
                  {
                    name: 'Expansion Tanks',
                    __typename: 'Category'
                  },
                  {
                    name: 'Instantaneous',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electronic Boards',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Specialty Plumbing Valves',
                children: [
                  {
                    name: 'Hydrants/Flow Control Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pressure Reducing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Wye Strainers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Relief Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Thermostatic Mixing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hose Bibbs',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dielectric Unions',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sillcocks',
                    __typename: 'Category'
                  },
                  {
                    name: 'Automatic Control Valves',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Import Valves',
                children: [
                  {
                    name: 'Ball Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Check Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gate Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Balancing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Butterfly Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Globe Valves',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Domestic Valves',
                children: [
                  {
                    name: 'Ball Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Check Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Butterfly Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gate Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Pumps',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Parts and Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Recirculating Pumps',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Backflow Preventers & Accessories',
                children: [
                  {
                    name: 'Backflow Preventers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Vacuum Breakers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Gaps',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Treatment',
                children: [
                  {
                    name: 'Water Filters',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Filter Parts & Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Softener Parts & Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Softeners',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Conditioners',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heaters - Residential',
                children: [
                  {
                    name: 'Gas',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electric',
                    __typename: 'Category'
                  },
                  {
                    name: 'Heat Pump',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heaters - Commercial',
                children: [
                  {
                    name: 'Electric',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gas',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'TBC',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sheet Metal Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Faucets, Fixtures Appliances',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Faucets, FiXtures & Appliances',
            children: [],
            __typename: 'Category'
          }
        ],
        __typename: 'ProductCategories'
      },
      __typename: 'Query'
    }
  }
};

type MockCategory = {
  request: {
    query: DocumentNode;
    variables: {
      engines?: string;
    };
  };
  result: {
    data: ProductCategoriesQuery | undefined;
  };
};
