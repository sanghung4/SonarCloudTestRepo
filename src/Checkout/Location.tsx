import { Box, Skeleton, Typography } from '@dialexa/reece-component-library';

import { CustomAddressInput } from 'Checkout/util/types';
import { Branch, ContractDetails, ErpAccount } from 'generated/graphql';
import { every } from 'lodash-es';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';
import { useMemo } from 'react';

type Props = {
  location?: ContractDetails | Branch | ErpAccount | CustomAddressInput;
  boldName?: boolean;
  includePhone?: boolean;
};

export default function Location(props: Props) {
  const {
    selectedAccounts: { billToErpAccount }
  } = useSelectedAccountsContext();

  /*
   * Memo
   */
  const address = useMemo(() => {
    // Nothing
    if (!props.location || every(props.location, (val) => !val)) {
      return undefined;
    }

    // When location type is `ContractDetails`
    if ('jobName' in props.location) {
      return {
        name: props.location.jobName,
        street1: props.location.accountInformation?.shipToAddress?.address1,
        street2: props.location.accountInformation?.shipToAddress?.address2,
        city: props.location.accountInformation?.shipToAddress?.city,
        state: props.location.accountInformation?.shipToAddress?.state,
        zip: props.location.accountInformation?.shipToAddress?.zip,
        phoneNumber: billToErpAccount?.phoneNumber ?? ''
      };
    }

    if ('name' in props.location) {
      return {
        name: props.location.name,
        street1: props.location.address1,
        street2: props.location.address2,
        city: props.location.city,
        state: props.location.state,
        zip: props.location.zip,
        phoneNumber: props.location.phone
      };
    }

    if ('companyName' in props.location) {
      return {
        name: props.location.companyName,
        street1: props.location.street1,
        street2: props.location.street2,
        city: props.location.city,
        state: props.location.state,
        zip: props.location.zip,
        phoneNumber: props.location.phoneNumber
      };
    }
  }, [props.location, billToErpAccount]);

  /**
   * Render
   */
  if (!address) {
    return (
      <Box data-testid="checkout-pickup-location-address-loading">
        <Typography variant="body1">
          <Skeleton width={150} />
        </Typography>
        <Typography variant="body1">
          <Skeleton width={200} />
        </Typography>
        <Typography variant="body1">
          <Skeleton width={100} />
        </Typography>
        <Typography variant="body1">
          <Skeleton width={200} />
        </Typography>
      </Box>
    );
  }

  return (
    <Box data-testid="checkout-pickup-location-address">
      {!!address?.name && (
        <Typography variant="body1" fontWeight={props.boldName ? 700 : 500}>
          {address.name}
        </Typography>
      )}
      {!!address?.street1 && (
        <Typography variant="body1">{address.street1}</Typography>
      )}
      {!!address?.street2 && (
        <Typography variant="body1">{address.street2}</Typography>
      )}
      {!!address?.city && !!address.state && !!address.zip && (
        <Typography variant="body1">
          {`${address.city}, ${address.state} ${address.zip}`}
        </Typography>
      )}
      {!!props.includePhone && !!address?.phoneNumber && (
        <Typography variant="body1">{address.phoneNumber}</Typography>
      )}
    </Box>
  );
}
