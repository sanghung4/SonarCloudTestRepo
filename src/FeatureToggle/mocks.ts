import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
