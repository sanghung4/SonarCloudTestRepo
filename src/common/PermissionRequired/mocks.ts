import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
