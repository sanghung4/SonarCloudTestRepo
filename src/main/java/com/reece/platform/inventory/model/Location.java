package com.reece.platform.inventory.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "pi_locations")
@Getter
@Setter
public class Location extends AuditableEntity {
    @ManyToOne
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @Column(name = "erp_location_id", nullable = false)
    private String erpLocationId;
}
