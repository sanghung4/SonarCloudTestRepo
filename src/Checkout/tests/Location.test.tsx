import { createMockBranches } from 'Branches/tests/mocks';
import Location from 'Checkout/Location';
import { mockErpAccount } from 'Checkout/tests/mocks/accounts.mocks';
import { mockCustomAddress } from 'Checkout/tests/mocks/deliveryMethodObject.mock';
import { mockData as mockContractDetails } from 'Contract/tests/mocks';
import { Branch } from 'generated/graphql';
import { render } from 'test-utils/TestWrapper';

describe('Checkout - Location', () => {
  it('should match snapshot with undefined props', () => {
    const { container } = render(<Location />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot as ContractDetails', () => {
    const { container } = render(
      <Location location={mockContractDetails} boldName includePhone />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot as Branch', () => {
    const mockBranch: Branch = {
      ...createMockBranches(1)[0],
      phone: '1234567890'
    };
    const { container } = render(
      <Location location={mockBranch} boldName includePhone />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot as ErpAccount', () => {
    const { container } = render(
      <Location location={mockErpAccount} boldName includePhone />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot as CustomAddressInput', () => {
    const { container } = render(
      <Location location={mockCustomAddress} boldName includePhone />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot without boldnam,e', () => {
    const { container } = render(
      <Location location={mockCustomAddress} includePhone />
    );
    expect(container).toMatchSnapshot();
  });
});
