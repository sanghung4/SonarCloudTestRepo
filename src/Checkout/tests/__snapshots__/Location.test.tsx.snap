// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout - Location should match snapshot as Branch 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="checkout-pickup-location-address"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-uzccd-MuiTypography-root"
    >
      Test 0
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      1234567890
    </p>
  </div>
</div>
`;

exports[`Checkout - Location should match snapshot as ContractDetails 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="checkout-pickup-location-address"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-uzccd-MuiTypography-root"
    >
      Steve Jobs
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      123 Fake St
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      Suite Fake
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      Fakinton, FA 99999
    </p>
  </div>
</div>
`;

exports[`Checkout - Location should match snapshot as CustomAddressInput 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="checkout-pickup-location-address"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-uzccd-MuiTypography-root"
    >
      Softlayer
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      123 Chimney Rock
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      #107
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      Houston, Texas 77012
    </p>
  </div>
</div>
`;

exports[`Checkout - Location should match snapshot as ErpAccount 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="checkout-pickup-location-address"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-uzccd-MuiTypography-root"
    >
      HG
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      123 S Gessner Rd
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      Suite 245
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      Houston, Texas 77472
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      832-965-9280
    </p>
  </div>
</div>
`;

exports[`Checkout - Location should match snapshot with undefined props 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="checkout-pickup-location-address-loading"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        style="width: 150px;"
      />
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        style="width: 200px;"
      />
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        style="width: 100px;"
      />
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
        style="width: 200px;"
      />
    </p>
  </div>
</div>
`;

exports[`Checkout - Location should match snapshot without boldnam,e 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="checkout-pickup-location-address"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
    >
      Softlayer
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      123 Chimney Rock
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      #107
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
    >
      Houston, Texas 77012
    </p>
  </div>
</div>
`;
